isa/kernel.org/instruction-set.rst

```diff
@@ -16,77 +30,141 @@ The eBPF calling convention is defined as:
+and destination registers, respectively, rather than the register number.
 
 Instruction classes
 -------------------
 
-The three LSB bits of the 'opcode' field store the instruction class:
-
-  =========  =====  ===============================
-  class      value  description
-  =========  =====  ===============================
-  BPF_LD     0x00   non-standard load operations
-  BPF_LDX    0x01   load into register operations
-  BPF_ST     0x02   store from immediate operations
-  BPF_STX    0x03   store from register operations
-  BPF_ALU    0x04   32-bit arithmetic operations
-  BPF_JMP    0x05   64-bit jump operations
-  BPF_JMP32  0x06   32-bit jump operations
-  BPF_ALU64  0x07   64-bit arithmetic operations
-  =========  =====  ===============================
+The encoding of the 'opcode' field varies and can be determined from
+the three least significant bits (LSB) of the 'opcode' field which holds
+the "instruction class", as follows:
+
+  =========  =====  ===============================  =======  =================
+  class      value  description                      version  reference
+  =========  =====  ===============================  =======  =================
+  BPF_LD     0x00   non-standard load operations     1        `Load and store instructions`_
+  BPF_LDX    0x01   load into register operations    1        `Load and store instructions`_
+  BPF_ST     0x02   store from immediate operations  1        `Load and store instructions`_
+  BPF_STX    0x03   store from register operations   1        `Load and store instructions`_
+  BPF_ALU    0x04   32-bit arithmetic operations     3        `Arithmetic and jump instructions`_
+  BPF_JMP    0x05   64-bit jump operations           1        `Arithmetic and jump instructions`_
+  BPF_JMP32  0x06   32-bit jump operations           3        `Arithmetic and jump instructions`_
+  BPF_ALU64  0x07   64-bit arithmetic operations     1        `Arithmetic and jump instructions`_
+  =========  =====  ===============================  =======  =================
+
+where 'version' indicates the first ISA version in which support for the value was mandatory.
 
 Arithmetic and jump instructions
 ================================
 
-For arithmetic and jump instructions (BPF_ALU, BPF_ALU64, BPF_JMP and
-BPF_JMP32), the 8-bit 'opcode' field is divided into three parts:
+For arithmetic and jump instructions (``BPF_ALU``, ``BPF_ALU64``, ``BPF_JMP`` and
+``BPF_JMP32``), the 8-bit 'opcode' field is divided into three parts:
 
   ==============  ======  =================
   4 bits (MSB)    1 bit   3 bits (LSB)
   ==============  ======  =================
-  operation code  source  instruction class
+  code            source  instruction class
   ==============  ======  =================
 
-The 4th bit encodes the source operand:
+code
+  the operation code, whose meaning varies by instruction class
+
+source
+  the source operand, which unless otherwise specified is one of:
```

Comment:
Changed "source operand" to "source operand location".  I didn't change the field name, since Linux has:

#define [BPF_SRC](https://elixir.bootlin.com/linux/latest/C/ident/BPF_SRC)([code](https://elixir.bootlin.com/linux/latest/C/ident/code))   (([code](https://elixir.bootlin.com/linux/latest/C/ident/code)) & 0x08)